src/core/public/saved_objects/saved_objects_client.ts

```diff
@@ -465,11 +465,7 @@ export class SavedObjectsClient {
    * uses `{response: {status: number}}`.
    */
   private savedObjectsFetch(path: string, { method, query, body }: HttpFetchOptions) {
-    return this.http.fetch(path, { method, query, body }).catch(err => {
-      const kfetchError = Object.assign(err, { res: err.response });
-      delete kfetchError.response;
-      return Promise.reject(kfetchError);
-    });
+    return this.http.fetch(path, { method, query, body });
```

Comment:
__note:__ discussed with @rudolf that's no longer necessary since BWC is maintained within HttpFetchError itself.